runtime/onert/core/include/ir/train/TrainingInfo.h

```diff
@@ -0,0 +1,68 @@
+
+/*
+ * Copyright (c) 2023 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __ONERT_IR_TRAIN_TRAINING_INFO_H__
+#define __ONERT_IR_TRAIN_TRAINING_INFO_H__
+
+#include <memory>
+
+#include "ir/train/OptimizerCode.h"
+#include "ir/train/OptimizerInfo.h"
+#include "ir/train/LossInfo.h"
+
+namespace onert
+{
+namespace ir
+{
+namespace train
+{
+
+class TrainingInfo final
+{
+public:
+  TrainingInfo() : _loss_info(), _optimizer_info(), _batch_size(0), _epoch(0) {}
+  // named constructor : create a default valid TrainingInfo
+  static std::unique_ptr<TrainingInfo> createDefaultInfo();
```

Comment:
Q: Why did you use the named constructor?  

A: I think it is confusing that some `ir` default-constructors make an invalid one, but sometimes not.  
So, I prefer to give a named constructor, to show the caller that this `ir` is filled with default 'valid' value.